DigiMeSDK/Core/Classes/DMEClient.m

```diff
@@ -302,30 +303,10 @@ - (void)getFileWithId:(NSString *)fileId completion:(FileContentCompletionBlock)
         __strong __typeof(DMEClient *)strongSelf = weakSelf;
         
         NSError *error;
-        
-        NSDictionary *json = [NSJSONSerialization JSONObjectWithData:data options:0 error:&error];
         NSData *decryptedData = [CADataDecryptor decrypt:data error:&error];
         
-        NSString *compression = json[@"compression"];
-        
-        CAFile *file;
-        
-        if (!error)
-        {
-            NSData *fileData;
-            
-            if ([compression isEqualToString:@"brotli"])
-            {
-                // Decompress data before serving to deserialiser.
-                fileData = [DMECompressor decompressData:decryptedData usingAlgorithm:DMECompressionAlgorithmBrotli];
-            }
-            else
-            {
-                fileData = decryptedData;
-            }
-        
-            file = [CAFile deserialize:fileData fileId:fileId error:&error];
-        }
+        NSData *fileData = [DMEDataUnpacker unpackData:decryptedData fromRootData:data];
```

Comment:
We are now ignoring possible `nil` results from decryption and unpacking.
Suggest:
```
CAFile *file;
if (decryptedData)
{
     NSData *fileData = [DMEDataUnpacker unpackData:decryptedData fromRootData:data];
    if (!fileData)
    {
        error = [NSError sdkError:SDKErrorInvalidData];
    }
else
{
file  = [CAFile deserialize:fileData fileId:fileId error:&error];
}
}
```